agglayer/grpc/conversion_to_agglayer.go

```diff
@@ -0,0 +1,298 @@
+			v.Mainnet.ProofLeafMer,
+			v.Mainnet.ProofGerL1Root,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("grpcClaimDataToAgglayer. error converting Mmainnet proofs: %w", err)
+		}
+		l1feaf, err := grpcL1LeafToAgglayer(v.Mainnet.L1Leaf)
+		if err != nil {
+			return nil, fmt.Errorf("grpcClaimDataToAgglayer. error converting Mainnet L1 leaf: %w", err)
+		}
+		return &agglayertypes.ClaimFromMainnnet{
+			ProofLeafMER:     proofs[0],
+			ProofGERToL1Root: proofs[1],
+			L1Leaf:           l1feaf,
+		}, nil
+	case *v1types.ImportedBridgeExit_Rollup:
+		proofs, err := grpcMerkleProofsToAgglayer(
+			v.Rollup.ProofLeafLer,
+			v.Rollup.ProofLerRer,
+			v.Rollup.ProofGerL1Root,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("grpcClaimDataToAgglayer. error converting Roolup proofs: %w", err)
+		}
+		l1feaf, err := grpcL1LeafToAgglayer(v.Rollup.L1Leaf)
+		if err != nil {
+			return nil, fmt.Errorf("grpcClaimDataToAgglayer. error converting Roolup L1 leaf: %w", err)
+		}
+		return &agglayertypes.ClaimFromRollup{
+			ProofLeafLER:     proofs[0],
+			ProofLERToRER:    proofs[1],
+			ProofGERToL1Root: proofs[2],
+			L1Leaf:           l1feaf,
+		}, nil
+	default:
+		return nil, fmt.Errorf("unknown claim type: %T", v)
+	}
+}
+
+func grpcMerkleProofsToAgglayer(proofs ...*v1types.MerkleProof) ([]*agglayertypes.MerkleProof, error) {
+	result := make([]*agglayertypes.MerkleProof, len(proofs))
+	for i, proof := range proofs {
+		converted, err := grpcMerkleProofToAgglayer(proof)
+		if err != nil {
+			return nil, fmt.Errorf("error converting merkle proof at index %d: %w", i, err)
+		}
+		result[i] = converted
+	}
+	return result, nil
+}
+
+func grpcMerkleProofToAgglayer(proof *v1types.MerkleProof) (*agglayertypes.MerkleProof, error) {
+	if proof == nil {
+		return nil, fmt.Errorf("grpcMerkleProofToAgglayer. proof is nil. %w", ErrNilCertificate)
+	}
+	if proof.Root == nil || proof.Siblings == nil {
+		return nil, fmt.Errorf("grpcMerkleProofToAgglayer. proof has nil Root or Siblings. %w", ErrNilCertificate)
+	}
+	if len(proof.Siblings) != int(treetypes.DefaultHeight) {
+		return nil, fmt.Errorf("grpcMerkleProofToAgglayer. proof has invalid number of siblings: %d. Expected: %d",
+			len(proof.Siblings), treetypes.DefaultHeight)
+	}
+	siblings := [treetypes.DefaultHeight]common.Hash{}
+	for i, sibling := range proof.Siblings {
+		siblings[i] = common.BytesToHash(sibling.Value)
+	}
+
+	return &agglayertypes.MerkleProof{
+		Root:  common.BytesToHash(proof.Root.Value),
+		Proof: siblings,
+	}, nil
+}
+
+func grpcL1LeafToAgglayer(l1Leaf *v1types.L1InfoTreeLeafWithContext) (*agglayertypes.L1InfoTreeLeaf, error) {
+	if l1Leaf == nil {
+		return nil, fmt.Errorf("grpcL1LeafToAgglayer. l1 leaf is nil. %w", ErrNilCertificate)
+	}
+	if l1Leaf.Rer == nil || l1Leaf.Mer == nil {
+		return nil, fmt.Errorf("grpcL1LeafToAgglayer. l1 leaf has nil Rer or Mer. %w", ErrNilCertificate)
+	}
+	var inner *agglayertypes.L1InfoTreeLeafInner
+	if l1Leaf.Inner != nil {
```

Comment:
I think `Inner` should never be nil. We should probably introduce a check for it.